### PR TITLE
feat(sdk.v2): Add HTML and Markdown artifact types

### PR DESCRIPTION
**Pull request by chensun**
**Description of your changes:**
Reference: #3970, #5798

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

**Comment by Bobgy** [comment]
/lgtm
thanks for adding the new types that will unblock @zijianjoy 's work

**Comment by Bobgy** [comment]
@zijianjoy can you link the PR to be part of the tracking issue

File: sdk/python/kfp/v2/compiler_cli_tests/test_data/pipeline_with_various_io_types.py [hunk 1]
```diff
@@ -60,6 +62,7 @@
 - {name: input_e, type: Datasets}
 - {name: input_f, type: Some arbitrary type}
 - {name: input_g, type: {GcsPath: {data_type: TSV}}}
+- {name: input_h, type: HTML}
```
Comment (zijianjoy) [inline sdk/python/kfp/v2/compiler_cli_tests/test_data/pipeline_with_various_io_types.py:65]: Can we also add the Markdown type input here?

**Comment by zijianjoy** [comment]
Thank you @chensun for making the PR!

Good point @Bobgy , I have added the associated issues in this PR's description.

File: sdk/python/kfp/v2/compiler_cli_tests/test_data/pipeline_with_various_io_types.py (hunk 1 above)
Comment (chensun) [reply sdk/python/kfp/v2/compiler_cli_tests/test_data/pipeline_with_various_io_types.py:65]: Not necessarily need to capture all the types we have in this test. 
But `type_utils_test.py` has UTs for the complete list of artifact types.

Comment (zijianjoy) [reply sdk/python/kfp/v2/compiler_cli_tests/test_data/pipeline_with_various_io_types.py:65]: I see, thanks Chen for confirming!

**Pull request merged**